nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -389,8 +421,25 @@ in
         zfsSupport = true;
       };
 
-      environment.etc."zfs/zed.d".source = "${packages.zfsUser}/etc/zfs/zed.d/";
-      environment.etc."zfs/zpool.d".source = "${packages.zfsUser}/etc/zfs/zpool.d/";
+      services.zfs.zed.settings = {
+        ZED_EMAIL_PROG = mkDefault "${pkgs.mailutils}/bin/mail";
+      };
+
+      environment.etc = let
+        zedDFiles = "${packages.zfsUser}/etc/zfs/zed.d";
+      in genAttrs
+        (map
+          (file: "zfs/zed.d/${file}")
+          (filter
+            (file: if file == "zed.rc" then false else true)
+            (builtins.attrNames (builtins.readDir zedDFiles))
```

Comment:
This needs IFD, which is not really allowed in nixpkgs. It can lead to problems with cross comp and remote builders. So a filelist is still needed unfortunately.